### PR TITLE
chore: upgrade to stable2407

### PR DESCRIPTION
**Pull request by AlexD10S**
With this template, we’re taking a different approach compared to our usual workflow for other templates. 
Directly upgraded to a stable version. 
The `psvm -v "stable2407"` command was not used due to various compilation errors; instead, importing directly from the GitHub repository and the latest in https://github.com/polkadot-evm/frontier which is `stable2407`.

### References used to compare
Base Parachain: https://github.com/r0gue-io/base-parachain/compare/polkadot-v1.9.0...polkadot-stable2409
https://github.com/OpenZeppelin/polkadot-runtime-templates/pull/299
https://github.com/polkadot-evm/frontier/pull/1504

### Testing
To test everything works, I followed the OpenZeppelin tutorial to deploy a "Hello World" smart contract: https://docs.openzeppelin.com/substrate-runtimes/2.0.2/guides/testing_with_zombienet#step_2_deploy_a_solidity_smart_contract
I added two accounts with balances in the `chain_spec` file to simplify the testing process above.

### CODE REVIEW

**Review by chungquantin — COMMENTED** [review]
Is it possible to base this PR on the another PR that covers the file refactoring? It is bit hard to see which changes are for `stable2409`.

File: node/src/service.rs [hunk 1]
```diff
@@ -1,77 +1,56 @@
...
-    BuildNetworkParams, CollatorSybilResistance, DARecoveryProfile, StartRelayChainTasksParams,
-};
-use cumulus_primitives_core::{
-    relay_chain::{CollatorPair, ValidationCode},
-    ParaId,
+    BuildNetworkParams, CollatorSybilResistance, DARecoveryProfile, ParachainHostFunctions,
+    StartRelayChainTasksParams,
 };
+use cumulus_primitives_core::relay_chain::ValidationCode;
+use cumulus_primitives_core::{relay_chain::CollatorPair, ParaId};
```
Comment (chungquantin) [inline node/src/service.rs]:
```suggestion
use cumulus_primitives_core::{
    relay_chain::{CollatorPair, ValidationCode},
    ParaId,
};
```
This should combine with the import above

**Review by chungquantin — APPROVED** [review]
- [x] Compiled and block produced successfully.
- [ ] Have you tried to deploy a Solidity contract? I tried `evm.create()` and it keeps loading forever. Maybe I did something wrong, just want to confirm.

**Review by al3mart — APPROVED** [review]
I was able to build the project and deploy a contract without problems with the provided steps.
Here a screenshots as proof :)

image

```
forge create --rpc-url http://localhost:40751 --private-key 0x5fb92d6e98884f76de468fa3f6278f8807c48bebc13595d45af5bdc4da702133 HelloWorld.sol:HelloWorld
[⠊] Compiling...
[⠒] Compiling 1 files with Solc 0.8.28
[⠊] Installing Solc version 0.8.28
[⠢] Successfully installed Solc 0.8.28
[⠆] Solc 0.8.28 finished in 935.67ms
Compiler run successful!
Deployer: 0xf24FF3a9CF04c71Dbc94D0b566f7A27B94566cac
Deployed to: 0x970951a12F975E6762482ACA81E57D5A2A4e73F4
Transaction hash: 0x6f3dc7259690144f48b2142380b0757debd1dabe1e7a46fdd966a3ea1441b36b
```

**Review by chungquantin — APPROVED** [review]
LGTM!

**Pull request merged**